plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -13,6 +13,81 @@ import { getByAge, UUIDT } from '../../utils/utils'
 
 type TeamCache<T> = Map<TeamId, [T, number]>
 
+function detectPropertyDefinitionTypes(isNumerical: boolean, value: any, key: string) {
+    let propertyType: PropertyType | null = null
+    let propertyTypeFormat: PropertyTypeFormat | null = null
+
+    /**
+     * Auto detecting unix timestamps is tricky. It's hard to know what is a big number or ID and what is a timestamp
+     *
+     * This tries to detect the most likely cases.
+     *
+     * * Numbers or Numeric Strings
+     * * That are either ten digits (seconds since unix epoch), or 13 digits (milliseconds since unix epoch),
+     * * or ten digits with numbers after the decimal place (whole seconds since unix epoch and fractions of a second)
+     * * where the property key includes either time or timestamp
+     *
+     * ten digits of seconds since epoch runs between Sep 09 2001 and Nov 20th 2286
+     *
+     * These are some representations from a variety of programming languages
+     *
+     * Python
+     * >>> datetime.now().timestamp()
+     * 1641477529.234715
+     *
+     * Ruby
+     * puts Time.now.to_i
+     * 1641477692
+     *
+     * Node JS
+     * console.log(Date.now())
+     * 1641477753371
+     *
+     * Java
+     * System.out.println(LocalDateTime.now().toEpochSecond(ZoneOffset.UTC));
+     * 1641478115
+     *
+     * SQL Lite
+     * select strftime('%s', 'now')
+     * 1641478347
+     *
+     * @param propertyValue
+     */
+    const detectUnixTimestamps = (propertyValue: string) => {
+        if (
+            (key.toLowerCase().indexOf('timestamp') > -1 || key.toLowerCase().indexOf('time') > -1) &&
```

Comment:
Looks like we can cover the same with just `key.toLowerCase().includes('time')`? It's also not entirely clear why `propertyValue` is a function argument, while `key` is grabbed from the parent scope, maybe the function could be parameter-less and just do `propertyValue.toString()`?

Reply:
Does it make sense to group everything `/^(\d{10}|\d{13})(\.\d*)?$/` under `unix_timestamp`? Should we instead use something like `unix_seconds` for one, and `unix_milliseconds` for the other?

Reply:
ClickHouse currently only parses 9 or 10 digit unix timestamps

https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions/#parsedatetimebesteffort

So even with a 13 digit timestamp we'll be throwing away the fraction of a second the final 3 digits represent

There's no harm in separating them at detection though